docs/development.md

```diff
@@ -0,0 +1,31 @@
+# Development
+This document contains some additional information and help regarding developing `mgmt`. Useful tools, conventions, etc.
+
+Be sure to read [quick start guide](docs/quick-start-guide.md) first.
+
+## Testing
+This project has both unit tests in the form of Go tests and integration tests using shell scripting.
+
+Native Golang tests are preferred over tests written in our shell testing framework. Please see https://golang.org/pkg/testing/ for more information.
+
+To run all tests:
+
+```
+make test
+```
+
+There is a library of quick and small integration tests for Language or YAML related things, check out `test/shell/`. Adding a test is as easy as copying one of the files in `test/shell/` and adapting it.
```

Comment:
These can be relative links with `[blah](url)`

s/Language or/the language and/